src/main/java/lotto/domain/game/LottoGame.java

```diff
@@ -4,22 +4,24 @@
 import java.util.Map;
 import lotto.domain.lotto.LottoTicket;
 import lotto.domain.lotto.Money;
-import lotto.domain.result.GameResult;
 import lotto.domain.lotto.Purchase;
+import lotto.domain.lotto.WinningTicket;
+import lotto.domain.result.GameResult;
 
 public class LottoGame {
 
-  private final LottoTicket winningTicket;
+  private final WinningTicket winningTicket;
 
-  public LottoGame(LottoTicket winningTicket) {
+  public LottoGame(WinningTicket winningTicket) {
     this.winningTicket = winningTicket;
   }
 
   public GameResult check(Purchase purchase) {
     Money winningPrize = new Money(0);
     Map<Rank, Integer> ranks = new HashMap<>();
     for (LottoTicket ticket : purchase.getTickets()) {
-      Rank rank = Rank.valueOf(winningTicket.countMatchingNumbers(ticket));
+      Rank rank = Rank.valueOf(winningTicket.countMatchingNumbers(ticket),
+          winningTicket.containsBonus(ticket));
```

Comment:
```suggestion
      Rank rank = winningTicket.match(ticket);;
```
위와 같이 winningTicket이 등수를 결정하도록 구현하면 어떨까?